### PR TITLE
✨ Defaulting webhooks should always set label from Spec.ClusterName

### LINKED ISSUE #2438: Defaulting webhooks should always set label from Spec.ClusterName
We need to make sure that defaulting webhooks for all our type that have a `Spec.ClusterName` field, have logic to default the Cluster Label to the value in Spec.ClusterName.

/kind feature
/milestone v0.3.0
/priority important-soon

### PR DESCRIPTION
**Pull request by prankul88**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2438

### CODE REVIEW

**Comment by ncdc** [comment]
/hold
pending logic correction to set the label, not the ObjectMeta.ClusterName

**Comment by ncdc** [comment]
@prankul88 will you have time to update this?

**Comment by prankul88** [comment]
@ncdc yes will update it today.
@JoelSpeed Thanks

**Comment by vincepri** [comment]
@prankul88 I rebased this PR and fixed the logic, this should be ready to go!

/assign @ncdc @JoelSpeed

File: api/v1alpha3/machinedeployment_webhook.go [hunk 1]
```diff
@@ -134,6 +134,10 @@ func PopulateDefaultsMachineDeployment(d *MachineDeployment) {
 		d.Spec.Template.Labels = make(map[string]string)
 	}
 
+	if len(d.ClusterName) == 0 {
+		d.ClusterName = d.Spec.ClusterName
+	}
```
Comment (JoelSpeed) [inline api/v1alpha3/machinedeployment_webhook.go]: This one is still doing the old logic

Comment (vincepri) [reply api/v1alpha3/machinedeployment_webhook.go]: done thanks!

**Pull request merged**